hazelcast/src/main/java/com/hazelcast/scheduledexecutor/impl/DistributedScheduledExecutorService.java

```diff
@@ -359,4 +375,60 @@ private void sendBatch(int partitionId, String name, List<ScheduledExecutorMerge
             invoke(SERVICE_NAME, operation, partitionId);
         }
     }
+
+    private class StatsLoggerThread extends HazelcastManagedThread {
+        private final Map<String, MutableInteger> tasksSizes = new HashMap<String, MutableInteger>();
+        private final ILogger logger = nodeEngine.getLogger(DistributedScheduledExecutorService.class);
+        private final IPartitionService partitionService = nodeEngine.getPartitionService();
+        private final long logDelayMillis;
+        private volatile boolean shutdown;
+
+        public StatsLoggerThread(int logDelaySeconds) {
+            this.logDelayMillis = TimeUnit.SECONDS.toMillis(logDelaySeconds);
+        }
+
+
+        private void shutdown() {
+            shutdown = true;
+            interrupt();
+        }
+
+        @Override
+        public void run() {
+            while (!shutdown) {
+                try {
+                    Thread.sleep(logDelayMillis);
+                } catch (InterruptedException e) {
+                    continue;
+                }
+
+                tasksSizes.clear();
+                for (ScheduledExecutorPartition partition : partitions) {
+                    if (partition == null) {
+                        continue;
+                    }
+
+                    if (partitionService.getPartitionOwner(partition.partitionId()) != nodeEngine.getThisAddress()) {
+                        // skip non primary partitions.
+                        continue;
+                    }
+
+                    for (Map.Entry<String, ScheduledExecutorContainer> entry : partition.containers.entrySet()) {
+                        String name = entry.getKey();
+                        ScheduledExecutorContainer container = entry.getValue();
+                        MutableInteger tasksSize = tasksSizes.get(name);
+                        if (tasksSize == null) {
+                            tasksSize = new MutableInteger();
+                            tasksSizes.put(name, tasksSize);
+                        }
+                        tasksSize.value += container.tasks.size();
```

Comment:
Why old_value + current_value? If it was 1 task in the previous log, and now we have two tasks in the container, we will log 3?

Reply:
An iteration over all the partitions is needed and for the same ScheduledExecutor we could run into multiple ScheduledExecutorContainer instances (for each partition would see one).